### ISSUE TITLE
What are the versions of the project pkg requirement?

### ISSUE THREAD

**Issue by RuifMaxx** [hongzimao/decima-sim#40]
I get some warning when running your code with tensorflow 1.14 in windows such as

> WARNING:tensorflow:From D:\DeepRL\decima-sim-master\actor_agent.py:19: The name tf.train.AdamOptimizer is deprecated. Please use tf.compat.v1.train.AdamOptimizer instead.

> WARNING:tensorflow:From test.py:20: The name tf.set_random_seed is deprecated. Please use tf.compat.v1.set_random_seed instead.

> WARNING:tensorflow:From test.py:30: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.
2021-10-10 23:22:55.372925: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX AVX2
WARNING:tensorflow:From D:\DeepRL\decima-sim-master\actor_agent.py:39: The name tf.placeholder is deprecated. Please use tf.compat.v1.placeholder instead.
WARNING:tensorflow:From D:\DeepRL\decima-sim-master\gcn.py:30: The name tf.sparse_placeholder is deprecated. Please use tf.compat.v1.sparse_placeholder instead.
WARNING:tensorflow:From D:\DeepRL\decima-sim-master\tf_op.py:34: The name tf.variable_scope is deprecated. Please use tf.compat.v1.variable_scope instead.
WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8E30C08>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8E30C08>>: AttributeError: module 'gast' has no attribute 'Index'
WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8E30188>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8E30188>>: AttributeError: module 'gast' has no attribute 'Index'
WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8E1F088>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8E1F088>>: AttributeError: module 'gast' has no attribute 'Index'
WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8DFDD48>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8DFDD48>>: AttributeError: module 'gast' has no attribute 'Index'
WARNING:tensorflow:From D:\DeepRL\decima-sim-master\actor_agent.py:225: calling softmax (from tensorflow.python.ops.nn_ops) with dim is deprecated and will be removed in a future version.
Instructions for updating:
dim is deprecated, use axis instead
WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8F43408>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8F43408>>: AttributeError: module 'gast' has no attribute 'Index'
WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8FAFB48>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8FAFB48>>: AttributeError: module 'gast' has no attribute 'Index'
WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8F87748>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8F87748>>: AttributeError: module 'gast' has no attribute 'Index'
WARNING:tensorflow:Entity <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8F55A48>> could not be transformed and will be executed as-is. Please report this to the AutgoGraph team. When filing the bug, set the verbosity to 10 (on Linux, `export AUTOGRAPH_VERBOSITY=10`) and attach the full output. Cause: converting <bound method Dense.call of <tensorflow.python.layers.core.Dense object at 0x0000014DF8F55A48>>: AttributeError: module 'gast' has no attribute 'Index'
WARNING:tensorflow:From D:\DeepRL\decima-sim-master\actor_agent.py:96: calling reduce_sum_v1 (from tensorflow.python.ops.math_ops) with keep_dims is deprecated and will be removed in a future version.
Instructions for updating:
keep_dims is deprecated, use keepdims instead
WARNING:tensorflow:From D:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\ops\math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where
WARNING:tensorflow:From D:\DeepRL\decima-sim-master\actor_agent.py:159: The name tf.train.Saver is deprecated. Please use tf.compat.v1.train.Saver instead.

>WARNING:tensorflow:From D:\ProgramData\Anaconda3\envs\tensorflow\lib\site-packages\tensorflow\python\training\saver.py:1276: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.

So what your pkg requirement of this code?

**Comment by hongzimao** [comment]
Those are deprecation warnings for some variable naming. Should still be fine running the code. IIRC we used tensorflow 1.12 at the time.

**Comment by RuifMaxx** [comment]
Thank you very much

**Comment by jahidhasanlinix** [comment]
I was trying to run this code after installing all the pkgs correctly, but I am getting this while running train.py, any suggestions on how to resolve it. Thank you.

$ python3 train.py 
Using backend: pytorch

**Comment by hongzimao** [comment]
This is weird. This repo is built on top of tensorflow. Don't know how pytorch is triggered. Do you have any more output information when running the code?

**Comment by jahidhasanlinix** [comment]
No, I did not get any output. I am not sure why, I am just having trouble with running this code so far. Just working on it to make it run. will update you once it is working. Apart from that if you have any suggestions please let me know so I can work on it. And thank you for your response too.

**Comment by wwww6555** [comment]
你好，我想问一下代码要求的tensorflow版本是什么呢
